test/terra/backends/qasm_simulator/qasm_unitary_gate.py

```diff
@@ -54,6 +57,29 @@ def test_random_unitary_gate(self):
         self.assertSuccess(result)
         self.compare_counts(result, circuits, targets, delta=0.05 * shots)
 
+    def test_random_unitary_gate_with_permutations(self):
+        """Test simulation with random unitary gate with permutations."""
+        all_permutations = list(itertools.permutations([0, 1, 2]))
+        unitary_matrix = random_unitary(8, seed=8)
+        n = 3
+        shots = 2000
+        qr = QuantumRegister(n, 'qr')
+        cr = ClassicalRegister(n, 'cr')
+        regs = (qr, cr)
+    
+        for perm in all_permutations:
+            circuit = QuantumCircuit(*regs)
+            circuit.unitary(unitary_matrix, perm)
+            circuit.barrier(qr)
+            circuit.measure(qr, cr)
```

Comment:
... and then
```
circuit.measure(range(n), range(n))
```
And similarly for the barrier in the previous line

Reply:
done